incremental-font-transfer/src/glyph_keyed.rs

```diff
@@ -98,20 +104,31 @@ pub(crate) fn apply_glyph_keyed_patch(
     Ok(font_builder.build())
 }
 
-fn table_tag_list(glyph_patches: &[GlyphPatches]) -> BTreeSet<Tag> {
-    glyph_patches
+fn table_tag_list(glyph_patches: &[GlyphPatches]) -> Result<BTreeSet<Tag>, PatchingError> {
+    for patches in glyph_patches {
+        if patches
+            .tables()
+            .iter()
+            .zip(patches.tables().iter().skip(1))
+            .any(|(prev_tag, next_tag)| next_tag <= prev_tag)
+        {
+            return Err(PatchingError::InvalidPatch(
+                "Duplicate or unsorted table tag.",
```

Comment:
Not really the fault of this PR but I want to register my dislike for using strings as error types; using distinct variants for each failure case forces you to actually enumerate the failure cases, and also makes it easier to handle particular failure cases with a match statement.

Reply:
So in the case of the InvalidPatch error type the callers handling of the error isn't going to change depending on how specifically the patch was invalid, the string here is to just help with debugging to provide some additional info about why we're rejecting it.